plugins/kubernetes-backend/src/kubernetes-auth-translator/AzureIdentityKubernetesAuthTranslator.ts

```diff
@@ -31,11 +42,23 @@ export class AzureIdentityKubernetesAuthTranslator
       clusterDetails,
     );
 
-    const credentials = new DefaultAzureCredential();
+    if (!this.accessToken || this.tokenExpired()) {
+      this.accessToken = await this.tokenCredential.getToken(aksScope);
+
+      if (!this.accessToken) {
+        throw new Error('Unable to retrieve Azure token');
+      }
+    }
 
-    // TODO: can we cache this? It's inneficiant to get a new token every time
-    const accessToken = await credentials.getToken(aksScope);
-    clusterDetailsWithAuthToken.serviceAccountToken = accessToken.token;
+    clusterDetailsWithAuthToken.serviceAccountToken = this.accessToken.token;
     return clusterDetailsWithAuthToken;
   }
+
+  private tokenExpired(): boolean {
+    if (!this.accessToken) return true;
+
+    // Set tokens to expire 2 minutes before its actual expiry time
+    const expiresOn = this.accessToken.expiresOnTimestamp - 2 * 60 * 1000;
```

Comment:
Is `expiresOnTimestamp` based on local time + expiry? There's a risk that a drifted clock could cause some issues here otherwise

Reply:
From my understanding that's coming from MS server, not client local time. `The access token's expiration timestamp in milliseconds, UNIX epoch time.` Source: https://docs.microsoft.com/en-us/javascript/api/@azure/core-auth/accesstoken?view=azure-node-latest

And `Date.now()`: The static Date.now() method returns the number of milliseconds elapsed since January 1, 1970 00:00:00 UTC.